### ISSUE TITLE
Implicit dependency on joiner makes the plugin incompatible with Rails 3.2

### ISSUE THREAD

**Issue by Androc** [abahgat/redmine_didyoumean#69]
Hello,

I just tried to install this plugin on this configuration :
```
Environment:
  Redmine version                2.6.1.stable
  Ruby version                   1.9.3-p194 (2012-04-20) [i486-linux]
  Rails version                  3.2.21
  Environment                    production
  Database adapter               Mysql2
```

## Problem

The installation of the plugin failed because of the dependency to `ActiveRecord 4.2.0` while Rails depends on `ActiveRecord 3.2.21`.

After taking a looking at the `Gemfile.lock` I saw that the problem comes from `joiner` which, since his version 0.3.0 (https://rubygems.org/gems/joiner/versions/0.3.0), depends on `Rails 4.1` or above.

## Solution

So, as dependency on joiner for this plugin and `thinking-sphinx` is `(~> 0.2.0)`, I just fix my problem this way :
* `rm Gemfile.lock`
* add `gem 'joiner', '~>0.2.0'` in Gemfile
* `bundle install`

If I am not wrong, you should make this modification to keep this plugin compatible with Rails 3.2.

**Comment by korin** [comment]
I use this plugin with
```
Environment:
  Redmine version                2.6.0.stable
  Ruby version                   1.9.3-p448 (2013-06-27) [x86_64-linux]
  Rails version                  3.2.21
  Environment                    production
  Database adapter               Mysql2
```
by default redmine do not care about Gemfile.lock file from plugin. How you install deps?

**Comment by Androc** [comment]
Gemfile.lock is not the problem, Gemfile.lock is the file which "gave" me the solution.

I saw that it was the joiner gem which asks a too recent version of ActiveRecord.

Could you please take a look at your Gemfile.lock and give me the joiner version you have ?

**Comment by korin** [comment]
in redmine Gemfile.lock
```
    joiner (0.2.0)
      activerecord (>= 3.1.0, < 4.1.0)
    thinking-sphinx (3.1.2)
      activerecord (>= 3.1.0)
      builder (>= 2.1.2)
      innertube (>= 1.0.2)
      joiner (>= 0.2.0)
      middleware (>= 0.1.0)
      riddle (>= 1.5.11)
```

**Comment by Androc** [comment]
Are you telling me that you installed the plugin fews days ago and you did not have the problem ?
The first version 0.3 of joiner was released on 2014/04/19.

Maybe you already installed the gem before and never had to update it.

**Comment by korin** [comment]
what you mean by "first version 0.3"?
I see all versions on [rubygems](https://rubygems.org/gems/joiner/versions)

To make sure everything is working I downloaded fresh [redmine](http://www.redmine.org/releases/redmine-2.6.1.tar.gz) and [plugin](https://github.com/abahgat/redmine_didyoumean/archive/master.zip)
I put plugin into redmine/plugins dir and from redmine dir execute `bundle install`.
No complains. Everything wa installed smoothly.

[bundle install result](https://gist.github.com/korin/b8419c1424944fa2173d)
[Gemfile.lock](https://gist.github.com/korin/5cef790bbaadb12b5424)

**Comment by Androc** [comment]
By "first version 0.3" I spoke about "the first version of the 0.3.X serie". The 0.3.0.

The problem does not come from the plugin itself but from the implicit dependency of joiner.

Before you make your test, uninstall joiner because, reading your log of bundle install, I see `Using joiner 0.2.0` which means you already have one.

If you don't have joiner, it will install you the last one leading to my problem.

**Comment by korin** [comment]
fully fresh install
https://gist.github.com/korin/b0a6d53f984cbf45ba7e

**Comment by Androc** [comment]
Apparently you don't have the problem. But it is strange, you don't have the latest version of gems.
For example, I have `rake 10.4.2` while you have 10.1.1.

I am too new to Ruby/Redmine to understand why the bundler don't try to install you the latest version of joiner if you don't precize it to not do it.

All what I know is that it was my problem was this dependency and I fixed it.

Maybe if I am the only one with this problem (and as I have the solution), we can close this issue.

**Comment by Androc** [comment]
Does bundle outdated` tells you your joiner is out dated ?

**Comment by korin** [comment]
yes, and many other gems too

**Comment by Androc** [comment]
It appears that from where you execute `bundle update` is important.

I uninstalled joiner (`gem uninstall joiner`).

I deleted Gemfile.lock of didyoumean plugin.

I ran `bundle update` from my redmine's root path ==> it installed the joiner 0.2.0.

I ran `bundle update` from the redmine_didyoumean directory, it updated joiner to 0.3.4.

So I suppose I did that when installed the plugin and that is why I had problem.

Thanks for your help.
